onnxruntime/core/providers/dml/DmlExecutionProvider/src/TensorDesc.cpp

```diff
@@ -212,15 +212,15 @@ gsl::span<const uint32_t> TensorDesc::GetStrides() const
 
 void TensorDesc::SetStrides(gsl::span<const uint32_t> strides)
 {
-    m_bufferTensorDesc.Strides = strides.empty() ? nullptr : strides.data();
```

Comment:
>Strides = strides.empty() ? nullptr : strides.data(); [](http://example.com/codeflow?start=23&length=53)

This is a bug o_o and potential use after free. We should not persist a pointer to a transient local from the caller.